code/modules/wiremod/components/utility/router.dm

```diff
@@ -0,0 +1,71 @@
+/**
+ * # Router Component
+ *
+ * Writes one of multiple inputs to one of multiple outputs.
+ */
+/obj/item/circuit_component/router
+	display_name = "Router"
+	display_desc = "Don't know how to wire up your circuit? This lets the circuit decide.\n\nThe input indicated by the Input input will be written to the output indicated by the Output input. Got it? Indices out of range will wrap around."
+	circuit_flags = CIRCUIT_FLAG_INPUT_SIGNAL|CIRCUIT_FLAG_OUTPUT_SIGNAL
+
+	/// Which ports to connect.
+	var/datum/port/input/nin
+	var/datum/port/input/nout
+
+	/// How many ports to have.
+	var/input_port_amount = 4
+	var/output_port_amount = 4
+
+	/// Current type of the ports
+	var/current_type
+
+	/// The ports to route.
+	var/list/datum/port/input/ins
+	var/list/datum/port/output/outs
+
+/obj/item/circuit_component/router/populate_options()
+	var/static/component_options = list(
+		PORT_TYPE_ANY,
+		PORT_TYPE_STRING,
+		PORT_TYPE_NUMBER,
+		PORT_TYPE_LIST,
+		PORT_TYPE_ATOM,
+	)
+	options = component_options
+
+/obj/item/circuit_component/router/Initialize()
+	. = ..()
+	current_type = current_option
+	nin = add_input_port("Input", PORT_TYPE_NUMBER, default = 1)
+	nout = add_input_port("Output", PORT_TYPE_NUMBER, default = 1)
```

Comment:
These names are too vague and will be incredibly misleading and hard to reference.